src/index.js

```diff
@@ -29,12 +32,37 @@ const queryClient = new QueryClient({
   },
 })
 
+const oidcConfig = {
+  authority: GlobalConfiguration.OIDC_AUTHORITY_URL,
+  client_id: GlobalConfiguration.AWS_CLIENT_ID,
+  redirect_uri: GlobalConfiguration.OAUTH2_REDIRECT_URL,
+  userStore: new WebStorageStateStore({ store: window.localStorage }),
+  onSigninCallback: () => {
+    // redirect to original location
+    const url = window.sessionStorage.getItem(ORIGINAL_DESTINATION)
+    if (url) {
+      window.sessionStorage.removeItem(ORIGINAL_DESTINATION)
+      window.location.replace(url)
+    } else {
+      // remove url params to complete the login
+      window.history.replaceState({}, document.title, window.location.pathname)
+      window.location.reload()
+    }
+  },
+}
+
+if (GlobalConfiguration.END_SESSION_URL) {
+  oidcConfig.metadataSeed = {
+    end_session_endpoint: GlobalConfiguration.END_SESSION_URL,
```

Comment:
Cognito has a special logout url that isn't advertized in the metadata, so we have to configure it manually